0003/README.md

```diff
@@ -0,0 +1,116 @@
+---
+id: 0003
+title: Centralized Logging on Tinkerbell
+status: POC Done, Code change ready
+authors: Chitrabasu Khare <chitrabasukhare89@gmail.com>
+---
+
+## Summary:
+
+To have a centralized logging framework for Tinkerbell. 
+
+## Overview: 
+
+- Tinkerbell will have the feature of centralized logging. 
+- By default, the logging feature would be supported by rsyslog. 
+- Logging feature should be implemented in such a way that a user can modify the type of logging (rsyslog or fluentbit) as per his needs. This may require the user to make additional changes on his end to enable that feature. 
+
+## Goals: 
+
+Goals cover the feature Details:
+- rsyslog is unix based package to capture the logs in a connected network. This entails having centralized logging using rsyslog. 
+- rsyslog is based on the top of syslog. 
+- rsyslog reads the log in syslog daemon service.
+- rsyslog will run on a logging server. Description of this logging server will be provided as a part of logging configuration.
+- Using some configuration files as rules, the user can define the logs he/she is interested in and can direct them to a defined file.
+- These files are core files which define how rsyslog centralized logging would work. 
+- Since this implementation uses daemon services, it is necessary to run containerized tink services in root privileges. 
+- All core tink services (including worker) would be using rsyslog implementation only and syslog as the log driver. It has been done to ensure logging of core services is correct and consistent. 
+- docker-compose file has the logging configurations for tink services on provisioner. For tink-worker changes will be osie at workflow-helper.sh
+- Action containers will have the logging as rsys or other (if user-defined).
+- tls and Log rotation support would be added. 
+
+
+## Progress:
+
+This implementation of rsyslog has been verified with POC and been discussed in-depth within the Packet and Infracloud team.  
+
+## Suggestions:
+
+- We want to ensure at a basic minimum there is a centralized logging framework which exists for tinkerbell. In order to have a generic implementation of configuring log-drivers below are some proposals.
+
+1) How to make action containers run with logging drivers.
+
+a) Place the logging details inside the workflow template. 
```

Comment:
From the perspective of user experience: I think that adding it to the template is easy to explain and surface in documentation. Which is not quite the same as the easiest or best for actual user experience, but just another perspective on getting it in front of users.

Reply:
@DailyAlice 

yes, having in the template is a good option. From the perspective of usage is where we are getting a mixed feeling. The template is more focused on workflow and logging is more focused on the infrastructure level, but again logging is for actions containers itself. So at one hand where it seems good, on the other hand, it seems like we should also look for some other options as well.

Reply:
Should we have a "default" logging strategy to set at the `tink` layer with the possibility of overriding it as described?! my expectation is that 99% of the time logging will go to the same place

Reply:
yes, ryslog will be the default logging structure. This point is more inclined to user needs if he is willing to modify the log-drives or log files location.